src/Tabs/Tabs.tsx

```diff
@@ -63,29 +66,37 @@ function reducer(state: State, action: Action): State {
  * In contrast to the antd Tabs component, this implementation does
  * not require tab panels to be direct children of tabs.
  */
-export function Tabs(props: TabsProps) {
-  const [state, dispatch] = useReducer(reducer, { tabs: [] });
+export function Tabs<TTabID = number | string>(props: TabsProps<TTabID>) {
+  const [state, dispatch] = useReducer<Reducer<State<TTabID>, Action<TTabID>>>(
+    reducer,
+    {
+      tabs: [],
+    },
+  );
 
   const tabsContextProps = useMemo(
-    (): TabsContextProps => ({
+    (): TabsContextProps<TTabID> => ({
       tabs: state.tabs,
       registerTab: (newTab) => {
         dispatch({ type: 'registerTab', newTab });
       },
-      unregisterTab: (tabId) => {
-        dispatch({ type: 'unregisterTab', tabId });
+      unregisterTab: (tabID) => {
+        dispatch({ type: 'unregisterTab', tabID });
       },
       onSelected:
         props.onSelected ??
-        ((tabId) => {
-          dispatch({ type: 'onSelected', tabId });
+        ((tabID) => {
+          dispatch({ type: 'onSelected', tabID });
         }),
-      activeTabId: props.activeTabId ?? state.activeTabId,
+      activeTabID: props.activeTabID ?? state.activeTabID,
     }),
-    [state.tabs, state.activeTabId, props.onSelected, props.activeTabId],
+    [state.tabs, state.activeTabID, props.onSelected, props.activeTabID],
   );
 
   return (
+    // TODO how to fix this generics issue?
+    // https://hipsterbrown.com/musings/musing/react-context-with-generics/
```

Comment:
As it is a TS-problem, ideally it should be solved with TS methods, here override. 
The second solution creates an additional JS function.

Reply:
Agreed, I ignored the mismatch here and documented why this problem exists.